include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -230,15 +230,27 @@ def ReadLValueOp : MooreOp<"read_lvalue", [SameOperandsAndResultType]> {
   }];
 }
 
+def ReadOp : MooreOp<"read"> {
+  let summary = "Like llvm.load";
+  let description = [{
+    //TODO:
+  }];
+  let arguments = (ins Arg<RefType, "", [MemRead]>:$input);
+  let results = (outs UnpackedType:$result);
+  let assemblyFormat = [{
+    $input attr-dict `:` type($input) `->` type($result)
+  }];
+}
+
```

Comment:
I think you need a `TypesMatchWith` constraint on the op here similar to `VariableOp` (just not optional), which takes the result type and wraps it in a ref type for the input. That would also allow you to omit the `type($input) ->` part, and it would add a verifier to the op which makes sure that the input is `ref<T>` and the result just `T`.